HW_TypeScript_Primitive_literal_types_and_object_typing.ts

```diff
@@ -0,0 +1,194 @@
+
+  get name(): string {
+    return this._name;
+  }
+}
+
+class Level {
+
+  _groups: Group[] = [];
+  _name: string;
+  _description: string;
+
+  constructor(name: string, description: string) {
+    this._name = name;
+    this._description = description;
+  }
+
+  addGroup(group: Group) {
+    this._groups.push(group);
+  }
+
+  removeGroup(groupName: string): void {
+    this._groups = this._groups.filter(group => group.directionName !== groupName);
+  }
+}
+
+class Group {
+
+  _area: Area | null = null;
+  _status: string | null = null;
+  _students: Student[] = [];
+  _directionName: string;
+  _levelName: string;
+
+  static STATUS_ACTIVE: string = 'active';
+  static STATUS_INACTIVE: string = 'inactive';
+
+  constructor(directionName: string, levelName: string) {
+    this._directionName = directionName;
+    this._levelName = levelName;
+  }
+
+  addStudent(student: Student): void {
+    this._students.push(student);
+  }
+
+  removeStudent(studentName: string): void {
+    this._students = this._students.filter(student => student.fullName !== studentName);
+  }
+
+  setStatus(status: string): void {
+    if (status === Group.STATUS_ACTIVE || status === Group.STATUS_INACTIVE) {
+      this._status = status;
+    } else {
+      throw new Error('Invalid status');
+    }
+  }
+
+  showPerformance(): Student[] {
+    return [...this._students].sort((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+  }
+
+  get directionName(): string {
+    return this._directionName;
+  }
+
+  get levelName(): string {
+    return this._levelName;
+  }
+
+  get status(): string | null {
+    return this._status;
+  }
+
+  get students(): Student[] {
+    return this._students;
+  }
+
+  set area(area: Area | null) {
+    this._area = area;
+  }
+
+  get area(): Area | null {
+    return this._area;
+  }
+}
+
+class Student {
+
+  _firstName: string;
+  _lastName: string;
+  _birthYear: number;
+  _grades: { workName: string; mark: number }[] = [];
+  _visits: { lesson: string; present: boolean }[] = [];
+
+  constructor(firstName, lastName, birthYear) {
+    this._firstName = firstName;
+    this._lastName = lastName;
+    this._birthYear = birthYear;
+  }
+
+  get fullName(): string {
+    return `${this._lastName} ${this._firstName}`;
+  }
+
+  set fullName(value: string) {
+    [this._lastName, this._firstName] = value.split(' ');
+  }
+
+  get age(): number {
+    return new Date().getFullYear() - this._birthYear;
+  }
+
+  getPerformanceRating(): number {
+    const gradeValues = Object.values(this._grades);
+
+    if (!gradeValues.length) return 0;
+
+    const averageGrade = this._grades.reduce((sum: number, grade: { workName: string; mark: number }) => sum + grade.mark, 0) / this._grades.length;
+    const attendancePercentage = (this._visits.filter(visit => visit.present === true).length / this._visits.length) * 100;
+
+    return (averageGrade + attendancePercentage) / 2;
+  }
+
+  setGrade(workName: string, mark: number) {
```

Comment:
Your realization is not a set accessor, which means that you should explicitly add a type annotation for the function return type.